Merge some speakers

h-lame opened the pull request:
I spotted some speaker issues with wrong attribution or duplicate versions, so this PR fixes a bunch.

TBH I just eyeballed the [speakers page](https://www.rubyvideo.dev/speakers) and spotted a few discrepancies.  I suspect there's a bunch more if you go digging (particularly with things like how people typically style their names vs longer or shorter versions).

Probably easiest to look commit by commit in the off chance you disagree with any particular changes (e.g. I have removed people from `speakers.yml` if I've merged them - maybe that's not right?).

marcoroth commented on data/euruko/euruko-2016/videos.yml:
```diff
@@ -74,19 +74,29 @@
-    - Krzysztof (Christopher) Wawer # Validations (23:39)
-    - Demir Zekić # The Importance of Teaching and Metoring (28:53)
-    - Jan Lelis # Whirly - The Friendly Terminal Spinner (35:25)
+    - Max Gorin
+    - Johanna Lang
+    - Dayana Mick
+    - Kinga Kalinowska
+    - Izabela Komorek
+    - Melanie Keatley
+    - Stephanie Nemeth
+    - Krzysztof (Christopher) Wawer
+    - Demir Zekić
+    - Jan Lelis
```
Thanks for adding the lightning talks into the description!

Can we keep the comment here? For the case where we want to implement #157 it would be great to have all the lightning talks in the same comment format so we can parse/search&replace them in one go!

marcoroth commented on data/rocky-mountain-ruby/rocky-mountain-ruby-2012/videos.yml:
```diff
@@ -65,7 +65,7 @@
     - Elain Marina
     - Jim Denton
-    - Tom Frei
+    - Thomas Frey
```
Nice catch!

marcoroth commented on data/goruco/goruco-2017/videos.yml:
```diff
@@ -51,9 +51,9 @@
+  raw_title: 'GORUCO 2017: Developer Productivity Engineering by Pan Thomakos'
   speakers:
-    - Panayiotis Thomakos
+    - Pan Thomakos
```
We have the concept of "Canonical Speakers". I think we should setup one for "Panayiotis Thomakos" that sets the canonical to "Pan Thomakos" .

marcoroth commented on data/speakers.yml:
```diff
@@ -5913,13 +5913,6 @@
   website: ""
   slug: james-duncan-davidson
   canonical_slug:
-- name: James Edward Gray
-  twitter: ""
-  github: JEG2
-  bio: ""
-  website: http://graysoftinc.com/
-  slug: james-edward-gray
-  canonical_slug:
```
Let's keep this one too and set the `canonical_slug` to `james-edward-gray-ii` so future references point to the right record.

marcoroth commented on data/speakers.yml:
```diff
@@ -10937,13 +10923,6 @@
   website: ablogaboutcode.com
   slug: pan-thomakos
   canonical_slug:
-- name: Panayiotis Thomakos
-  twitter: ""
-  github: ""
-  bio: ""
-  website: ""
-  slug: panayiotis-thomakos
-  canonical_slug:
```
Let's keep this one too and set the `canonical_slug` to `pan-thomakos` so future references point to the right record.

marcoroth commented on data/speakers.yml:
```diff
@@ -13915,20 +13885,6 @@
   website: http://tomdale.net
   slug: tom-dale
   canonical_slug:
-- name: Tom E Enebo
-  twitter: ""
-  github: ""
-  bio: ""
-  website: ""
-  slug: tom-e-enebo
-  canonical_slug:
```
Let's keep this one too and set the `canonical_slug` to `thomas-e-enebo` so future references point to the right record.

marcoroth reviewed (commented):
Thanks @h-lame, very good detective work! 🕵🏼

h-lame commented:
Thanks for the feedback @marcoroth, I've rebased those changes into the original commits to keep the entries in speakers.yml where they are usuful redirect and to keep the comments in the Euruko lightning talks (can't think why I removed them in the first place when I left them in the day 2 talks 🤷 ).  While I was there I also added a few more merges, this time I think I have a better understanding of when to remove someone from speakers.yml (typos) and when to keep with a redirect (different styling of the name).

marcoroth reviewed (approved):
Awesome, thank you so much @h-lame! 🙏🏼
